src/NexusMods.DataModel/Loadouts/LoadoutRegistry.cs

```diff
@@ -118,6 +120,8 @@ public Loadout Alter(LoadoutId id, string commitMessage, Func<Loadout, Loadout>
 
         _logger.LogInformation("Loadout {LoadoutId} altered: {CommitMessage}", id, commitMessage);
 
+        var marker = _markers.GetOrAdd(id, id => new LoadoutMarker(this, id));
+        marker.SetDataStoreId(newLoadout.DataStoreId);
```

Comment:
You make a good point, but in the next few weeks I'm rewriting this code when we move to the event sourcing system, so I'll leave it for now.

This is a good example of how the current Datamodel breaks down, however. Making a new version of an entity persists it to the database, but we don't have a good definition of when the app as a whole moves to the next version of the loadout. We've used immutable data, but haven't defined the "moving forward in time" very well. So we could be adding a mod to a loadout when it suddenly gets updated and we'd need to restart the process. Or what if we're ingesting files and then the loadout is modified mid process? 

This all gets solved in the event system because the loadouts are completely read-only and are never edited by individual threads, instead they issue commands like "Add a mod with these files", and that gets logged as an event then the loadout is only ever updated by a single thread, maintaining a consistent view of the model.